gpytorch/kernels/cylindrical_kernel.py

```diff
@@ -0,0 +1,136 @@
+        :attr:`radial_base_kernel` (gpytorch.kernel):
+            The base kernel for computing the radial kernel
+        :attr:`batch_size` (int, optional):
+            Set this if the data is batch of input data.
+            It should be `b` if :attr:`x1` is a `b x n x d` tensor. Default: `1`
+        :attr:`eps` (float):
+            Small floating point number used to improve numerical stability
+            in kernel computations. Default: `1e-6`
+        :attr:`param_transform` (function, optional):
+            Set this if you want to use something other than softplus to ensure positiveness of parameters.
+        :attr:`inv_param_transform` (function, optional):
+            Set this to allow setting parameters directly in transformed space and sampling from priors.
+            Automatically inferred for common transformations such as torch.exp or torch.nn.functional.softplus.
+
+    """
+    def __init__(
+        self,
+        num_angular_weights,
+        radial_base_kernel,
+        batch_size=1,
+        eps=1e-6,
+        angular_weights_prior=None,
+        alpha_prior=None,
+        beta_prior=None,
+        param_transform=softplus,
+        inv_param_transform=None,
+        **kwargs
+    ):
+        super().__init__(has_lengthscale=False, batch_size=batch_size)
+        self.num_angular_weights = num_angular_weights
+        self.radial_base_kernel = radial_base_kernel
+        self.eps = eps
+        self._param_transform = param_transform
+        self._inv_param_transform = _get_inv_param_transform(param_transform, inv_param_transform)
+        self.register_parameter(name="raw_angular_weights",
+                                parameter=torch.nn.Parameter(torch.zeros(batch_size, num_angular_weights)))
+        self.register_parameter(name="raw_alpha", parameter=torch.nn.Parameter(torch.zeros(batch_size)))
+        self.register_parameter(name="raw_beta", parameter=torch.nn.Parameter(torch.zeros(batch_size)))
+        if angular_weights_prior is not None:
+            self.register_prior(
+                "angular_weights_prior", angular_weights_prior,
+                lambda: self.angular_weights, lambda v: self._set_angular_weights(v)
+            )
+        if alpha_prior is not None:
+            self.register_prior(
+                "alpha_prior", alpha_prior, lambda: self.alpha, lambda v: self._set_alpha(v)
+            )
+        if beta_prior is not None:
+            self.register_prior(
+                "beta_prior", beta_prior, lambda: self.beta, lambda v: self._set_beta(v)
+            )
+
+    @property
+    def angular_weights(self):
+        return self._param_transform(self.raw_angular_weights)
+
+    @angular_weights.setter
+    def angular_weights(self, value):
+        self._set_param(value, 'angular_weights')
+
+    @property
+    def alpha(self):
+        return self._param_transform(self.raw_alpha)
+
+    @alpha.setter
+    def alpha(self, value):
+        self._set_param(value, 'alpha')
```

Comment:
The setters should apply the inverse transform to the value before initializing the raw parameter in `_set_param`